network_setup.sh

```diff
@@ -100,6 +100,30 @@ function check_connection {
 	fi
 }
 
+function check_ssh_host_keys() {
+	# Sometimes these files are empty after first boot and ssh won't work.
+	# This will check the files for size and regenerate them when one of them is empty
+	if
+		file_empty "/etc/ssh/ssh_host_ecdsa_key" || file_empty "/etc/ssh/ssh_host_ecdsa_key.pub" ||
+			file_empty "/etc/ssh/ssh_host_ed25519_key" || file_empty "/etc/ssh/ssh_host_ed25519_key.pub" ||
+			file_empty "/etc/ssh/ssh_host_rsa_key" || file_empty "/etc/ssh/ssh_host_rsa_key.pub"
+	then
+		echo "Regenerating ssh host keys and restarting sshd"
+		ssh-keygen -A
```

Comment:
Also for the other commands: do we need sudo? Not sure how this script will be run....

Reply:
The script is started by systemctl with privileges, so sudo is not required. Also quite a lot of other commands(nmcli) in that script need privileges when running, but are started without sudo.